src/lib/group.ts

```diff
@@ -0,0 +1,394 @@
+import { Scalar, withMessage } from './core.js';
+import { Field, FieldVar, isField } from './field.js';
+import { Bool, Snarky } from '../snarky.js';
+import { Field as Fp } from '../provable/field-bigint.js';
+import { Pallas } from '../bindings/crypto/elliptic_curve.js';
+
+export { Group };
+
+type FieldLike = FieldVar | Field | number | string | bigint;
+type GroupLike = {
+  x: FieldLike;
+  y: FieldLike;
+};
+
+/**
+ * An element of a Group.
+ */
+class Group {
+  x: Field;
+  y: Field;
+
+  /**
+   * The generator `g` of the Group.
+   */
+  static get generator() {
+    return new Group(Pallas.one.x, Pallas.one.y);
+  }
+
+  /**
+   * Coerces anything group-like to a {@link Group}.
+   */
+  constructor(g: {
+    x: FieldVar | Field | number | string | bigint;
```

Comment:
Using the overloaded constructor so that the new implementation is equivalent to the old OCaml one, which allowed both `(x, y)` as well as `{x, y}` as arguments. I am not using the `FieldLike` or `GroupLike` type from above so users arent confused by it and see "the actual type".